sdk/appconfiguration/app-configuration/src/secretReference.ts

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { JsonSecretReferenceValue } from "./internal/jsonModels";
+import { ConfigurationSetting, ConfigurationSettingParam } from "./models";
+
+/**
+ * content-type for the secret reference.
+ */
+export const secretReferenceContentType =
+  "application/vnd.microsoft.appconfig.keyvaultref+json;charset=utf-8";
+
+/**
+ * Necessary fields for updating or creating a new secret reference.
+ */
+export interface SecretReferenceValue {
+  /**
+   * Id for the secret reference.
+   */
+  secretId: string;
+}
+
+export const SecretReferenceHelper = {
+  /**
+   * Takes the ConfigurationSetting and returns the FeatureFlag.
+   */
+  fromConfigurationSetting: (
+    setting: ConfigurationSetting
+  ): ConfigurationSetting<SecretReferenceValue> => {
+    if (!isSecretReference(setting)) {
+      throw new Error("Not a SecretReference..");
+    }
+    let jsonSecretReferenceValue: JsonSecretReferenceValue;
+    try {
+      if (!setting.value || typeof setting.value !== "string") {
+        throw new Error("");
+      }
+      jsonSecretReferenceValue = JSON.parse(setting.value) as JsonSecretReferenceValue;
+    } catch (err) {
+      // best effort - if it doesn't deserialize properly we'll just throw
+      throw new Error("");
+    }
+
+    const featureflag: ConfigurationSetting<SecretReferenceValue> = {
+      ...setting,
+      value: { secretId: jsonSecretReferenceValue.uri }
+    };
+    return featureflag;
+  },
+  /**
+   * Takes the FeatureFlag (JSON) and returns a ConfigurationSetting (with the props encodeed in the value).
+   */
+  toConfigurationSettingParam: (
+    secretReference: ConfigurationSettingParam<SecretReferenceValue>
+  ): ConfigurationSettingParam => {
+    if (!secretReference.value) {
+      throw new Error("Value is not defined");
+    }
+
+    const jsonSecretReferenceValue: JsonSecretReferenceValue = {
+      uri: secretReference.value.secretId
+    };
+
+    let stringifiedValue: string;
+    try {
+      stringifiedValue = JSON.stringify(jsonSecretReferenceValue);
+    } catch (err) {
+      // best effort - if it doesn't serialize properly we'll just throw
+      throw new Error("");
```

Comment:
:eyes:

Reply:
When wouldn't it serialize properly?

Reply:
![image](https://user-images.githubusercontent.com/10452642/118196959-cc923400-b402-11eb-8189-ef93adeaaeed.png)

Reply:
Why don't we allow the TypeError to bubble up? Instead of catching it and rethrowing a different exception?

Reply:
> throw one with no error message 

I had added the message yesterday!

> I imagine these errors will be surfaced to the user when they call one of our APIs with an invalid object, so they'd have context about what object was passed.

> Why don't we allow the TypeError to bubble up? Instead of catching it and rethrowing a different exception?




Fine, I won't catch and throw.

Reply:
> Also, I don't see logging of errors. Are they being caught, logged, and rethrown somewhere else?

Looks like the logger is not imported in app-config, should I do it now?

Reply:
> I had added the message yesterday!

Sorry about that. I left these comments yesterday but must have missed the 'submit' button.

Reply:
> Looks like the logger is not imported in app-config, should I do it now?

App config doesn't have any logging at all right now? In that case can you create an issue to add logging to app config and we can track doing that later?

Reply:
Added https://github.com/Azure/azure-sdk-for-js/issues/15288